### PR TITLE
Gene tree improvements

### PR DESCRIPTION
**Pull request by imran-ebi**
This pull request is for two JIRA tickets:

1. 4340 - tweaks to the gene tree such as legend label fix, scaling the legend, and improving the highlight message (bug)
2. 4341 - tweaks to gene tree zmenu (enhancement)

### CODE REVIEW

**Comment by ens-st3** [comment]
Hi Imran - do you have a commit on ensembl-webcode for ENSWEB-4341 ?

**Comment by imran-ebi** [comment]
@ens-st3 Yes, it is in the branch with the same name. I will make a pull request for it too. The image's legend scaling issue is not resolved, however I will make another pull request once that fix is available.

**Comment by ens-ma7** [comment]
Hi Imran
can i have your sandbox URL where the fix is? 
Thanks
RIdwan

**Comment by imran-ebi** [comment]
@ens-ma7 Yes sure. The link to it is: http://ves-hx-78.ebi.ac.uk:9200/Chondrus_crispus/Gene/Compara_Tree/pan_compara?anc=12149838;db=core;g=CHC_1040;g1=Npun_R3807;r=Pt:155059-155592;t=CCP38213

File: modules/EnsEMBL/Web/Component/Gene/ComparaTree.pm [hunk 1]
```diff
@@ -50,9 +50,22 @@ sub content {
 
   my $leaves               = $tree->get_all_leaves;
   my $tree_stable_id       = $tree->tree->stable_id;
-  my $highlight_gene       = $self->param('g1');
+  my $highlight_gene       = $hub->param('g1');
+  my $highlight_status     = $hub->get_cookie_value('gene_tree_highlighting') || 'on'; # get the the highlight switch status from the cookie
+
+  # Set $highlight_gene to undefined if the highlight status is off. This is due to the module relying heavily on $highlight_gene to do the rendering based on the highlight status.
```
Comment (ens-ma7) [inline modules/EnsEMBL/Web/Component/Gene/ComparaTree.pm]: This is duplicated with line 64, you have the same code twice. can you please delete one of them.

File: modules/EnsEMBL/Web/ZMenu/Gene.pm [hunk 2]
```diff
@@ -34,17 +34,19 @@ sub _content {
...
-    link  => $hub->url({
-      type => 'Gene',
-      action => 'Summary',
-      g => $g,
-#     __clear => '1' #gets rid of the url variation data in order to avoid a bug                                      
-    })
-  });
+
+  if($zmenu_label) {
+    # $self->add_entry({
```
Comment (ens-ma7) [inline modules/EnsEMBL/Web/ZMenu/Gene.pm]: why do you have these line of code commented? If its not being used can you please remove it

**Comment by ens-ma7** [comment]
After reviewing with @imran, the following was suggested:
1) the switching off higlight links at the bottom of the image doesnt work
2) Make the selected gene a bit bigger (font-size: 9 or 10)
3)

**Pull request merged**